h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -0,0 +1,239 @@
+    ModelBuilder builder = ModelBuilder.make(parms._infogram_algorithm_parameters);
+    Model builtModel = (Model) builder.trainModel().get();
+    Scope.track_generic(builtModel);
+    TwoDimTable varImp = extractVarImp(parms._algorithm, builtModel);
+    String[] ntopPredictors = new String[parms._top_n_features];
+    String[] rowHeaders = varImp.getRowHeaders();
+    System.arraycopy(rowHeaders, 0, ntopPredictors, 0, parms._top_n_features);
+    return ntopPredictors;
+  }
+
+  public static TwoDimTable extractVarImp(InfogramParameters.Algorithm algo, Model model) {
+    switch (algo) {
+      case gbm : return ((GBMModel) model)._output._variable_importances;
+      case glm : return ((GLMModel) model)._output._variable_importances;
+      case deeplearning : return ((DeepLearningModel) model)._output._variable_importances;
+      case drf : return ((DRFModel) model)._output._variable_importances;
+      case xgboost : return ((XGBoostModel) model)._output._variable_importances;
+      default : return null;
+    }
+  }
+
+  /**
+   * This method will perform two functions:
+   * - if user only wants a fraction of the training dataset to be used for infogram calculation, we will split the
+   *   training frame and only use a fraction of it for infogram training purposes;
+   * - next, a new training dataset will be generated containing only the predictors in predictors2Use array. 
+   *
+   * @param parms
+   * @param sensitivePredictors
+   * @param dataFraction
+   * @param
+   * @return
+   */
+  public static Frame extractTrainingFrame(InfogramParameters parms, String[] sensitivePredictors, double dataFraction,
+                                           Frame trainFrame) {
+    if (dataFraction < 1) {  // only use a fraction training data for speedup
+      SplitFrame sf = new SplitFrame(trainFrame, new double[]{parms._data_fraction, 1-parms._data_fraction},
+              new Key[]{Key.make("train.hex"), Key.make("discard.hex")});
+      sf.exec().get();
+      Key[] ksplits = sf._destination_frames;
+      trainFrame = DKV.get(ksplits[0]).get();
+      DKV.remove(ksplits[1]); // discard unwanted portion
+    }
+    final Frame extractedFrame = new Frame(Key.make());
+    if (sensitivePredictors != null)
+      for (String colName : sensitivePredictors) // add sensitive features to Frame
+        extractedFrame.add(colName, trainFrame.vec(colName));
+
+    List<String> colNames = Arrays.asList(trainFrame.names());
+    if (parms._weights_column != null && colNames.contains(parms._weights_column))
+      extractedFrame.add(parms._weights_column, trainFrame.vec(parms._weights_column));
+    if (parms._offset_column != null && colNames.contains(parms._offset_column))
+      extractedFrame.add(parms._offset_column, trainFrame.vec(parms._offset_column));
+    if (parms._response_column != null && colNames.contains(parms._response_column))
+      extractedFrame.add(parms._response_column, trainFrame.vec(parms._response_column));
```

Comment:
you can probably just look through `parms.getNonPredictors()`

Reply:
Removed all this mess.

Reply:
I can still see it :)

Reply:
Done